sql/core/src/test/scala/org/apache/spark/sql/sources/BucketedReadSuite.scala

```diff
@@ -543,6 +551,68 @@ abstract class BucketedReadSuite extends QueryTest with SQLTestUtils {
     )
   }
 
+  test("SPARK-19122 Re-order join predicates if they match with the child's output partitioning") {
+    val bucketedTableTestSpec = BucketedTableTestSpec(
+      Some(BucketSpec(8, Seq("i", "j", "k"), Seq("i", "j", "k"))),
+      numPartitions = 1,
+      expectedShuffle = false,
+      expectedSort = false)
+
+    def testBucketingWithPredicate(
+        joinCondition: (DataFrame, DataFrame) => Column,
+        expectedResult: Option[Array[Row]]): Array[Row] = {
+      testBucketing(
+        bucketedTableTestSpecLeft = bucketedTableTestSpec,
+        bucketedTableTestSpecRight = bucketedTableTestSpec,
+        joinCondition = joinCondition,
+        expectedResult = expectedResult
+      )
+    }
+
+    // Irrespective of the ordering of keys in the join predicate, the query plan and
+    // query results should always be the same
```

Comment:
I think we don't need to test this here, this is an existing property of join implementation in Spark SQL, and should have already been well tested. Then we don't need to change `testBucketing`